### ISSUE TITLE
Prefer ipv6 for listening check in distcc.sh

### ISSUE THREAD

**Issue by steakhal** [whisperity/dotfiles#27]
```sh
# Returns true if $1 is listening on the local machine.
function check_tcp4port_listen {
    _dccsh_debug -n "Check port $1..."

    ss -lt4 | grep ":$1" &>/dev/null   # <----- HERE
    local R=$?

    _dccsh_debug -n "    is listening? "
    if [ $R -eq 0 ]; then
        _dccsh_debug "YES"
    else
        _dccsh_debug "NO"
    fi

    return $R
}
```

[source-code](https://github.com/whisperity/dotfiles/blob/ace18e744531596d488cb6fba7c0ce51c2fe17ae/packages/distcc/distcc.sh#L146)

For my setup, the `-lt6` works fine. But not the other way around.
Do you think we should change it?

**Comment by whisperity** [comment]
Sorry, none of the networks I'm on support IPv6 so I have the entire yadda disabled until ISPs get around upgrading their stack... Also, if we check for v6, then the function name has to be changed. I am worried about something like `:1234` matching an IPv6 address in the output. How does the output look for you? Maybe we need to explicitly regex match at the end of lines or something. A sample output would really help to see this.

**Comment by whisperity** [comment]
Okay, hopped onto a server and produced two output.

```
whisperity$ ss -lt4
State                    Recv-Q                    Send-Q                                       Local Address:Port                                        Peer Address:Port
LISTEN                   0                         5                                                  0.0.0.0:15001                                            0.0.0.0:*
LISTEN                   0                         20                                               127.0.0.1:smtp                                             0.0.0.0:*
LISTEN                   0                         5                                                  0.0.0.0:16670                                            0.0.0.0:*
LISTEN                   0                         128                                                0.0.0.0:ssh                                              0.0.0.0:*
whisperity$ ss -lt6
State                    Recv-Q                    Send-Q                                       Local Address:Port                                        Peer Address:Port
LISTEN                   0                         128                                                      *:16665                                                  *:*
LISTEN                   0                         20                                                   [::1]:smtp                                                [::]:*
LISTEN                   0                         128                                                      *:16669                                                  *:*
LISTEN                   0                         128                                                      *:15166                                                  *:*
LISTEN                   0                         128                                                   [::]:ssh                                                 [::]:*
```

Turns out `-lt6` would break for me if the listening socket is open IPv4-**only** (?) so a different approach is to be taken.

Odd. I guess in the end, we might assume the machine is just routeable and do not specify neither `-4` nor `-6`, and just check for listening sockets... And if we are already there, how about making the filter more robust...

```
whisperity$ ss -nlt | tail -n +2 | awk '{ print $4; }' | awk -F":" '{ print $NF; }'
15001
25
16670
34983
22
16665
25
16669
15166
22
```

 - `ss -n` so the ports are **always** numeric
 - `tail -n +2` cuts the first line ("Local Address ...")
 - Get 4th column, the one with the local address and port
 - Cut out all the `:` separated crap from it, and return the very last `:`-separated component, no matter the number. This should be the port, or `*`, but `*` as port would be very, very odd.

Now can `grep` in this without the fear mentioned in the previous comment.

**Comment by steakhal** [comment]
How should I proceed?

**Comment by whisperity** [comment]
@steakhal Two things... First, please send the results of `ss -lt4` and `ss -lt6` from a computer when you're actively trying to use the script (i.e. your SSH tunnels are up and running).
Second, replace the `ss -lt4 | grep ":$1" &>/dev/null` line with `ss -nlt | tail -n +2 | awk '{ print $4; }' | awk -F":" '{ print $NF; }' | grep ":$1" &>/dev/null` and check if your stuff still works.

If it does properly, I'll update the repository in a jiffy.

**Comment by whisperity** [comment]
Fixed in commit 7ab048d664a94e94b98af81940d5d645ec3e9eca. Previous comment contained a typo, you no longer need to grep the `:`.
